Support 2019.03

montjoie opened the pull request:
Hello, this is a PR which adds support for LAVA 2019.03 and all fixes found when working on it.

montjoie committed:
lava-master-base: 2019.03 needs python3-junit.xml

LAVA 2019.03 needs python3-junit.xml from stretch-backports.

montjoie committed:
lava-xxx-base: permit to install 2019.03

LAVA has changed its repository location.
Furthermore, it is now accessible via https which need additionnal
package to use it.

montjoie committed:
lava-xxx-base: do not install recommended packages

For limiting size of docker images, do not install recommended packages

montjoie committed:
lava-master: handle order of LAVA patchs

The current way to add patchs is non-determinist and could lead to
failure to build.
This patch made the order determinist by using sort

montjoie committed:
lava-slave: dont hide lavacli device-types list errors

montjoie committed:
lava-slave: fix build failure due to pxe on non-amd64

The package grub-efi-amd64-bin is only availlable on amd64.
Installing it on arm64 lead to a build failure.
This patch permits to install it even on arm64

montjoie committed:
lavalab-gen.py: add ALLOWED_HOSTS

Since 2019.03 , ALLOWED_HOSTS is mandatory in settings.conf.
Without it all requests are denied with code 500.

montjoie committed:
Use 2019.03

Since 2019.03 is ready, use it

montjoie committed:
lava-slave: Applying patch need the patch package

montjoie committed:
lava-master: warning when overwriting existing device-types

montjoie committed:
lava-slave: telnet is needed for ser2net

khilman commented on lavalab-gen.py:
```diff
@@ -149,13 +151,17 @@ def main():
         f_fqdn.close()
         fsettings = open("%s/settings.conf" % workerdir, 'w')
-        fsettings.write(template_settings_conf.substitute(cookie_secure=cookie_secure, session_cookie_secure=session_cookie_secure, lava_http_fqdn=lava_http_fqdn))
+        fsettings.write(template_settings_conf.substitute(cookie_secure=cookie_secure, session_cookie_secure=session_cookie_secure, lava_http_fqdn=lava_http_fqdn, allowed_hosts=allowed_hosts))
```
Looks like this is missing a default value:

```
$ ./lavalab-gen.sh
Handle master1

Traceback (most recent call last):
  File "./lavalab-gen.py", line 631, in <module>
    main()
  File "./lavalab-gen.py", line 165, in main
    f_fqdn.write(lava_http_fqdn)
UnboundLocalError: local variable 'lava_http_fqdn' referenced before assignment
```

montjoie committed:
lava-slave: PXE: remove a File Not Found error

biosdisk is not provided in default install and is not used, remove it.

montjoie committed:
lava-master: fix the applying of patchs against LAVA

Lava patch which contains device-type changes are ignored since final
device-types are already in /etc/lava-server/device-types/
So for fixing this, we apply lava-patch before device-type patch and
synchronize them.
